### ISSUE TITLE
Wrong start color-scheme if ssr true

### ISSUE THREAD

**Issue by GreyXor** [userquin/vuetify-nuxt-module#58]
Hello,
I want my site to automatically display the dark or light version of vuetify, this code won't work if ssr: true
so I coded this example  : https://stackblitz.com/github/greyxor/nuxt-vuetify-bug-scheme

The problem is that when the site opens, vuetify stay in a strange state, half of website is light, and other half is dark the light scheme (my system is in dark scheme)

![image](https://github.com/userquin/vuetify-nuxt-module/assets/79602273/cbc2e525-fdb3-4ed7-b28d-15356f6b3d45)

once on the site, if I change the scheme system setting to dark or light, then it works correctly.

![image](https://github.com/userquin/vuetify-nuxt-module/assets/79602273/32f8a171-09db-4f52-81ed-10c07dd75699)

Also if i disable nuxt SSR, then that fix the problem and theme is directly good at start of website

**Comment by userquin** [comment]
I guess you should configure dark/light themes. 

I've added `vuetifyOptions.theme.defaultTheme = 'dark'` and it will configure dark but since I have light in the browser, the theme is switched and the navigation drawer is white:

https://stackblitz.com/edit/github-rzg7of?file=nuxt.config.ts

**Comment by userquin** [comment]
I seems something related with Vuetify, v-app is not changing the theme (I've updated previous SB repro):

Dark: check v-app still with mainTheme and the navigation drawer with the mainDarkTheme:
![imagen](https://github.com/userquin/vuetify-nuxt-module/assets/6311119/2b681ec8-ca65-4760-9189-45c2b12cdfc0)

Light:
![imagen](https://github.com/userquin/vuetify-nuxt-module/assets/6311119/c20c91c5-2679-40f8-aa21-55d6b6eafa0f)

**Comment by userquin** [comment]
@GreyXor add this Nuxt plugin to your plugins directory, you have it in the SB repro:
```ts
// plugins/vuetify-ssr-theme.client.ts
export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.hook('vuetify:before-create', ({ vuetifyOptions }) => {
    if (vuetifyOptions.ssr) {
      const media = window.matchMedia('(prefers-color-scheme: dark)');

      function checkTheme() {
        useNuxtApp().$vuetify.theme.global.name.value = media.matches
          ? 'mainDarkTheme'
          : 'mainTheme';
      }

      nuxtApp.hook('app:suspense:resolve', () => {
        media.addEventListener('change', checkTheme);
        checkTheme();
      });
    }
  });
});
```

**Comment by GreyXor** [comment]
Thanks @userquin , the last solution is working as expected, but I don't really like the hook to `app:suspense:resolve`. Should we report issue to vuetify ?

**Comment by mubaidr** [comment]
Maybe you need to update `v-app` component too. Try this:

```vue
<VApp
    :theme="isDark ? 'dark' : 'light'"
  >
```

**Comment by userquin** [comment]
@mubaidr I guess it is  an hydration problem, the suspense hook will run after client hydration, if you dont have it the theme is the default one. The server will use the default configuration, and so we cannot solve the problem in the client without that hook, vuetify cannot handle this.

I Will try to find a solution for this with internal logic, maybe adding some composable for ssr, replacing useTheme to be nuxt  ssr aware.

**Comment by mubaidr** [comment]
> @mubaidr I guess it is an hydration problem, the suspense hook will run after client hydration, if you dont have it the theme is the default one. The server will use the default configuration, and so we cannot solve the problem in the client without that hook, vuetify cannot handle this.
> 
> I Will try to find a solution for this with internal logic, maybe adding some composable for ssr, replacing useTheme to be nuxt ssr aware.

Maybe this is temporary solution but it works. `isDark` above is provided by following composable: 

```ts
export function useCustomTheme() {
  const { $vuetify } = useNuxtApp()

  const isDark = useDark({
    valueDark: 'dark',
    valueLight: 'light',
    initialValue: 'light',
    onChanged: (dark: boolean) => {
      $vuetify.theme.global.name.value = dark ? 'dark' : 'light'
    },
  })

  const toggle = useToggle(isDark)

  return { isDark, toggle }
}

```

Wihtout this, `vuetify` does not behave well with `ssr` enabled.

**Comment by userquin** [comment]
@GreyXor @mubaidr should be fixed with the new features included in v0.5.10: check the SSR section in the documentation and the playground
